druid/src/core.rs

```diff
@@ -513,18 +517,26 @@ impl<T: Data, W: Widget<T>> WidgetPod<T, W> {
                 };
 
                 if let Some(change) = this_changed {
-                    let event = LifeCycle::FocusChanged(change);
-                    self.inner.lifecycle(ctx, &event, data, env);
+                    // Only send FocusChanged in case there's actual change
+                    if old != new {
+                        self.state.has_focus = change;
+                        let event = LifeCycle::FocusChanged(change);
+                        self.inner.lifecycle(ctx, &event, data, env);
+                    }
                     false
                 } else {
-                    old.map(|id| self.state.children.contains(&id))
-                        .or_else(|| new.map(|id| self.state.children.contains(&id)))
-                        .unwrap_or(false)
+                    self.state.has_focus = false;
+                    match (old, new) {
+                        (Some(old), _) if self.state.children.contains(old) => true,
+                        (_, Some(new)) if self.state.children.contains(new) => true,
+                        _ => false,
+                    }
                 }
             }
             LifeCycle::FocusChanged(_) => {
-                self.state.request_focus = None;
-                true
+                // We are a descendant of a widget that has/had focus.
```

Comment:
should we be clearing this somewhere else, now?

Reply:
There's nothing to clear because `request_focus` only gets set starting from the widget that does the request. This code here only executes for descendants of those widgets.

Reply:
ah I see and we were already clearing it above.